src/core/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -317,7 +317,15 @@ bool QgsGdalProvider::crsFromWkt( const char *wkt )
       CPLFree( pszProj4 );
 
       char *pszWkt = nullptr;
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,0,0)
+      const QByteArray multiLineOption = QStringLiteral( "MULTILINE=NO" ).toLocal8Bit();
```

Comment:
looking at the logic of the whole function, I don't get it. So we have as input... wkt, that we import as a OGRSpatialReferenceH object, then export it as WKT again, to finally pass it to QgsCoordinateReferenceSystem::fromWkt() ... What's the point :-) ? I suspect this might have to do with some normalization, but this looks really odd. Not even sure this is needed at all

Reply:
Ehh yeah 🤢 , I should have scrolled up and looked at the wider context here (I was originally just grepping for OSRExportToWkt). I'll refactor this stuff, it's gross.